File: src/main/java/repository/RepositoryInterface.java
package repository;

import domain.Entity;

public interface RepositoryInterface<ID,E extends Entity<ID>> {


    default E getOne(ID id){
        return null;
    }

    default Iterable<E> getAll(){
        return null;
    }

    default E add(E entity){
        return null;
    }

    default E delete(ID id){
        return null;
    }

    default E update(E entity){
        return null;
    }


}

File: persistence/RepositoryInterface.cs
using System;
using System.Collections.Generic;
using System.Text;
using LabMPP.domain;

namespace LabMPP.repository
{
    interface RepositoryInterface<ID,E> where E:Entity<ID>
    {
        E getOne(ID id);

        IEnumerable<E> getAll();

        E add(E entity);

        E delete(ID id);

        E update(E entity);
    }
}
